packages/coin-vet/src/sign.ts

```diff
@@ -0,0 +1,152 @@
+import { apdu, error, tx } from '@coolwallet/core';
+import * as txUtil from './utils/transactionUtil';
+import * as scriptUtil from './utils/scriptUtil';
+import * as types from './config/types';
+import { handleHex } from './utils/stringUtil';
+import { blake2b256 } from './vet/blake2b';
+
+const elliptic = require('elliptic');
+const ec = new elliptic.ec('secp256k1');
+const rlp = require('rlp');
+const blake2b = require('blake2b');
+
+/**
+ * Sign VeChain Transaction
+ */
+export async function signTransaction(
+  signTxData: types.signTxType,
+  publicKey: string
+): Promise<string> {
+  const { transaction, transport, addressIndex, appPrivateKey, appId, confirmCB, authorizedCB } = signTxData;
+
+  const preActions = [];
+
+  const { script, argument } = await scriptUtil.getScriptAndArguments(addressIndex, transaction);
+
+  const sendScript = () => apdu.tx.sendScript(transport, script);
+  preActions.push(sendScript);
+  const action = () => apdu.tx.executeScript(transport, signTxData.appId, signTxData.appPrivateKey, argument);
+
+  const signature = await tx.flow.getSingleSignatureFromCoolWallet(
+    transport,
+    preActions,
+    action,
+    false,
+    confirmCB,
+    authorizedCB,
+    true
+  );
+
+  console.log("vet signature: ", signature);
+
+  const { signedTx } = await apdu.tx.getSignedHex(transport);
+  console.log("signedTx: ", signedTx);
+
+  const rawTx = txUtil.getRawTx(transaction);
+  const rawData = rlp.encode(rawTx);
+  console.log("rawData: ", rawData.toString('hex'))
```

Comment:
Please remove console log in final version

Reply:
We have cleaned up the changes. Please verify. Thanks